cs/cs/VowpalWabbitJsonThreadedPrediction.cs

```diff
@@ -0,0 +1,34 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="VowpalWabbitJsonThreadedPrediction.cs">
+//   Copyright (c) by respective owners including Yahoo!, Microsoft, and
+//   individual contributors. All rights reserved.  Released under a BSD
+//   license as described in the file LICENSE.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace VW
+{
+    /// <summary>
+    /// Enables multi-threaded prediction by utilizing a pool of <see cref="VowpalWabbitJson"/> instances.
+    /// </summary>
+    public sealed class VowpalWabbitJsonThreadedPrediction : VowpalWabbitThreadedPredictionBase<VowpalWabbitJson>
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="VowpalWabbitThreadedPrediction{TExample}"/>.
+        /// </summary>
+        /// <param name="model">The model used by each pool instance.</param>
+        public VowpalWabbitJsonThreadedPrediction(VowpalWabbitModel model)
+            : base(model)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="VowpalWabbit{TExample}"/>.
+        /// </summary>
+        /// <param name="vw">The wrapped vw instance.</param>
+        sealed protected override VowpalWabbitJson InternalCreate(VowpalWabbit vw)
```

Comment:
The base class defines a protected method, thus the protected. The sealed here makes sure nobody can override the method, but since the class is already sealed it’s not needed.

Done.

From: Luong Hoang [mailto:notifications@github.com]
Sent: Wednesday, February 3, 2016 12:55 PM
To: JohnLangford/vowpal_wabbit <vowpal_wabbit@noreply.github.com>
Cc: Markus Cozowicz <marcozo@microsoft.com>
Subject: Re: [vowpal_wabbit] JSON & ASP.NET (#932)


In cs/cs/VowpalWabbitJsonThreadedPrediction.cs<https://github.com/JohnLangford/vowpal_wabbit/pull/932#discussion_r51759641>:

> +    public sealed class VowpalWabbitJsonThreadedPrediction : VowpalWabbitThreadedPredictionBase<VowpalWabbitJson>

> +    {

> +        /// <summary>

> +        /// Initializes a new instance of <see cref="VowpalWabbitThreadedPrediction{TExample}"/>.

> +        /// </summary>

> +        /// <param name="model">The model used by each pool instance.</param>

> +        public VowpalWabbitJsonThreadedPrediction(VowpalWabbitModel model)

> +            : base(model)

> +        {

> +        }

> +

> +        /// <summary>

> +        /// Creates a new instance of <see cref="VowpalWabbit{TExample}"/>.

> +        /// </summary>

> +        /// <param name="vw">The wrapped vw instance.</param>

> +        sealed protected override VowpalWabbitJson InternalCreate(VowpalWabbit vw)

"sealed protected" seems self-contradictory? :) maybe I'm not understanding what it means. The class is already sealed as well.

—
Reply to this email directly or view it on GitHub<https://github.com/JohnLangford/vowpal_wabbit/pull/932/files#r51759641>.